src/lib/y2network/type_detector.rb

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface_type"
+
+module Y2Network
+  # Detects type of given interface. New implementation of what was in
+  # @see Yast::NetworkInterfaces.GetType
+  class TypeDetector
+    class << self
+      # Finds type of given interface
+      #
+      # @param iface [String] interface name
+      #
+      # @return [Y2Network::InterfaceType, nil] type of given interface or nil if cannot be recognized
+      def type_of(iface)
+        type_by_sys(iface) || type_by_config(iface) || nil
+      end
+
+    private
+
+      include Yast::Logger
+
+      SYS_TYPE_NUMBERS = {
```

Comment:
any link or documentation for this magic numbers?

Reply:
i'm not aware of any ... taken from wicked sources